docs/api/implementation-guide.md

```diff
@@ -0,0 +1,67 @@
+---
+title: NuGet Server Implementation Guide
+description: Guidelines and recommendations to anyone implementing the NuGet Server API in their own package repository
+author: zivkan
+ms.author: zivkan
+ms.date: 07/29/2023
+ms.topic: conceptual
+---
+
+# NuGet Server Implementation Guide
+
+Prior to 2019, NuGet package repositories could copy the package metadata from the `nupkg`'s [`nuspec` file](../reference/nuspec.md), and return the results unmodified in [search](./search-query-service-resource.md) and [package metadata](./registration-base-url-resource.md) requests.
+Since then, NuGet has evolved to provide developers with a richer experience, and this requires package repositories to do additional work in order to provide the additional value to developers.
+
+## URL structure for private feeds
```

Comment:
One may misunderstand "private feed" as a "custom feed that's not nuget.org" instead of "feed that requires authentication". To clear up this confusion, I'd consider starting this paragraph with something like "For feeds that require authentication, ..."